truss/templates/shared/lazy_data_resolver.py

```diff
@@ -0,0 +1,70 @@
+from concurrent.futures import ThreadPoolExecutor
+from datetime import datetime
+from pathlib import Path
+from typing import Dict, List
+
+import pydantic
+import yaml
+
+try:
+    from shared.util import download_from_url_using_requests
+except ModuleNotFoundError:
+    from truss.templates.shared.util import download_from_url_using_requests
+
+LAZY_DATA_RESOLVER_PATH = Path("/bptr/bptr-manifest")
+NUM_WORKERS = 4
+
+
+class Resolution(pydantic.BaseModel):
+    url: str
+    expiry: datetime
+
+
+class BasetenPointer(pydantic.BaseModel):
```

Comment:
how will we keep these in sync with the definitions in the monorepo? we might want to add a note somewhere (maybe in the definitions in baseten_shared) that any updates need to be copied over

Reply:
Yeah this is a problem for us currently, let me make a note in the monorepo in the meantime – may want to think about publishing the `baseten_shared` package